### DISCUSSION TITLE
Multi-tenant app and granting consent

### DISCUSSION THREAD

**Discussion by cloudhal** [microsoft/hack-together#168] [Q&A]
So I have created a multi-tenant Blazor WASM app, which all works fine. However, when first logging in, user is presented with the box to add the app the tenant:
![image](https://user-images.githubusercontent.com/68559542/225297596-50e5ef7e-7be0-4363-a0c2-9483f064e315.png)
User is a global admin in the tenant, however, either clicking either Accept, or the Consent button, the app still does not load due to unauthorized error:
![image](https://user-images.githubusercontent.com/68559542/225297975-00d3f6a9-88fa-4beb-b5be-25f2b1e3cec5.png)

If I then go into the Enterprise App in Azure AD and click Consent, then it starts to work:

![image](https://user-images.githubusercontent.com/68559542/225298121-14b91124-e85e-4466-bdb8-8786a7a91cf9.png)

This is odd as it already says that permission has been grated through admin consent. Does anyone know why?

**Comment by waldekmastykarz** [comment]
You're having a NullReference somewhere in your code. Have you tried stepping through your code to see what's wrong?

**Reply by cloudhal** [reply]
Thanks Waldek, I may do (probably several), can a null reference cause this issue?

**Reply by waldekmastykarz** [reply]
I suspect it might be a timing issue, where propagating the consent takes a moment, but your code expects an instantaneous result which leads to the exception. So the exception itself is not causing the issue but could be a part of the broken end-user experience.
